### ISSUE TITLE
`--disable-uncorrectable` option does not insert todo comment as expected for multiline case

### ISSUE THREAD

**Issue by KessaPassa** [rubocop/rubocop#11706]
thank you for the fix to https://github.com/rubocop/rubocop/pull/11701
but, another issue similar to https://github.com/rubocop/rubocop/issues/11685 is there.

## Expected behavior

Rubocop should process

```ruby
ARRAY = %w[
  AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
].freeze
```
 
as

```ruby
# rubocop:todo Layout/LineLength
ARRAY = %w[
  AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
].freeze
# rubocop:enable Layout/LineLength
```
for "--disable-uncorrectable" option

## Actual behavior

it become infinite loop and I got

```ruby
# rubocop:todo Metrics/CollectionLiteralLength
ARRAY = %w[
  # rubocop:todo Layout/LineLength
  # rubocop:enable Layout/LineLength
  # rubocop:enable Layout/LineLength
  ...
  # rubocop:todo Layout/LineLength
  # rubocop:todo Layout/LineLength
  # rubocop:enable Layout/LineLength
  # rubocop:todo Layout/LineLength
  AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
# rubocop:enable Layout/LineLength
  # rubocop:todo Layout/LineLength
# rubocop:enable Layout/LineLength
  ...
# rubocop:enable Layout/LineLength
  # rubocop:todo Layout/LineLength
  # rubocop:todo Layout/LineLength
  # rubocop:enable Layout/LineLength
].freeze
# rubocop:enable Metrics/CollectionLiteralLength
```

## Steps to reproduce the problem

1. create `test_case.rb`
2. create`.rubocop.yml`
3. run `bundle exec rubocop -A --disable-uncorrectable test_case.rb`

test_case.rb
```ruby
ARRAY = %w[
  AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
].freeze
```

.rubocop.yml
```yml
Layout/LineLength:
  Max: 30
```

## RuboCop version

This happens with HEAD of master (4346f8ef4)

```
$ [bundle exec] rubocop -V
1.48.1 (using Parser 3.2.1.1, rubocop-ast 1.27.0, running on ruby 3.0.5) [x86_64-darwin22]
  - rubocop-performance 1.16.0
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.19.0
```

**Comment by nobuyo** [comment]
Ahh I probably made a mistake.
I think I should have used `overlaps?` instead of `crossing?` here:  https://github.com/rubocop/rubocop/blob/4346f8ef496599c1dfc53332df5116de735e3382/lib/rubocop/cop/autocorrect_logic.rb#L85
